packages/frontend/tests/acceptance/programs-test.js

```diff
@@ -88,6 +87,35 @@ module('Acceptance | Programs', function (hooks) {
     });
   });
 
+  module('User in multiple schools', function (hooks) {
+    hooks.beforeEach(async function () {
+      this.school1 = this.server.create('school');
+      this.school2 = this.server.create('school');
+      this.program1 = this.server.create('program', { school: this.school1 });
+      this.program2 = this.server.create('program', { school: this.school2 });
+      this.user = await setupAuthentication();
+    });
+
+    test('remember non-default school filter choice', async function (assert) {
+      await page.visit();
+
+      assert.strictEqual(currentURL(), '/programs');
+      assert.strictEqual(page.root.schoolFilter.schools.length, 2);
+
+      assert.strictEqual(page.root.schoolFilter.selectedSchool, '1');
+
+      await page.root.schoolFilter.select(2);
+      assert.strictEqual(page.root.schoolFilter.selectedSchool, '2');
+      assert.strictEqual(currentURL(), '/programs?school=2');
+
+      await click('.list tbody tr:nth-of-type(1) td:nth-of-type(1) a');
+      assert.strictEqual(currentURL(), '/programs/2');
+      await click('.backtolink a');
```

Comment:
This is is going to be a page object too, you'll have to import the one for the program and probably add this link there in the tree.

Reply:
Fixed in new PR https://github.com/ilios/frontend/pull/7942